File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/BatchTag.java
package uk.nhs.nhsx.keyfederation;

import uk.nhs.nhsx.core.ValueType;

public class BatchTag extends ValueType<BatchTag> {
        private BatchTag(String value) {
            super(value);
        }
        public static BatchTag of(String tag) {
            return new BatchTag(tag);
        }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/BatchTag.kt
package uk.nhs.nhsx.keyfederation

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class BatchTag private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<BatchTag>(::BatchTag)
}
